bot/exts/info/source.py

```diff
@@ -1,62 +1,75 @@
+from typing import Optional, Tuple
 
-from discord import Embed
-from discord.ext import commands
+from discord import Embed, Interaction, app_commands
+from discord.ext.commands import BadArgument, Cog, Command, ExtensionNotLoaded, HelpCommand
 
 from bot.bot import Bot
 from bot.constants import URLs
-from bot.converters import SourceConverter
+from bot.converters import SourceTransformer
 from bot.exts.info.tags import TagIdentifier
 
-SourceType = Union[commands.HelpCommand, commands.Command, commands.Cog, TagIdentifier, commands.ExtensionNotLoaded]
+SourceType = HelpCommand | Command | Cog | TagIdentifier | ExtensionNotLoaded | app_commands.Command | str
+HybridCommand = app_commands.Command | Command
 
 
-class BotSource(commands.Cog):
+class BotSource(Cog):
     """Displays information about the bot's source code."""
 
     def __init__(self, bot: Bot):
         self.bot = bot
 
-    @commands.command(name="source", aliases=("src",))
-    async def source_command(self, ctx: commands.Context, *, source_item: SourceConverter = None) -> None:
+    @app_commands.command(name="source")
+    async def source_command(
+        self,
+        interaction: Interaction,
+        *,
+        cog_command_or_tag: app_commands.Transform[SourceType, SourceTransformer] = None
+    ) -> None:
         """Display information and a GitHub link to the source code of a command, tag, or cog."""
-        if not source_item:
+        if not cog_command_or_tag:
             embed = Embed(title="Bot's GitHub Repository")
             embed.add_field(name="Repository", value=f"[Go to GitHub]({URLs.github_bot_repo})")
             embed.set_thumbnail(url="https://avatars1.githubusercontent.com/u/9919")
-            await ctx.send(embed=embed)
+            await interaction.response.send_message(embed=embed)
             return
 
-        embed = await self.build_embed(source_item)
-        await ctx.send(embed=embed)
+        embed = None
+        ephemeral = False
+        if isinstance(cog_command_or_tag, str):
+            description = f"**Unable to convert '{cog_command_or_tag}' to valid command, tag, or cog.**"
+            embed = Embed(description=description)
+            ephemeral = True
 
-    def get_source_link(self, source_item: SourceType) -> Tuple[str, str, Optional[int]]:
+        embed = await self.build_embed(cog_command_or_tag) if not embed else embed
+        await interaction.response.send_message(embed=embed, ephemeral=ephemeral)
+
+    def get_source_link(self, source_object: SourceType) -> Tuple[str, str, Optional[int]]:
         """
-        Build GitHub link of source item, return this link, file location and first line number.
+        Build GitHub link of source object, return this link, file location and first line number.
 
-        Raise BadArgument if `source_item` is a dynamically-created object (e.g. via internal eval).
+        Raise BadArgument if `source_object` is a dynamically-created object (e.g. via internal eval).
```

Comment:
Raise `CommandInvokeError` instead.

Reply:
Updated in https://github.com/python-discord/bot/pull/2446/commits/4601217e7c241f101b1cdf8b9779e4536e4dd4b5